simulator/cluster_compute_resource.go

```diff
@@ -350,6 +350,40 @@ func (c *ClusterComputeResource) ReconfigureComputeResourceTask(ctx *Context, re
 	}
 }
 
+func (c *ClusterComputeResource) MoveIntoTask(ctx *Context, req *types.MoveInto_Task) soap.HasFault {
+	task := CreateTask(c, "moveInto", func(*Task) (types.AnyType, types.BaseMethodFault) {
+		for _, ref := range req.Host {
+			host := ctx.Map.Get(ref).(*HostSystem)
+
+			if *host.Parent == c.Self {
+				return nil, new(types.DuplicateName) // host already in this cluster
+			}
+
+			switch parent := ctx.Map.Get(*host.Parent).(type) {
+			case *ClusterComputeResource:
+				if !host.Runtime.InMaintenanceMode {
+					return nil, new(types.InvalidState)
+				}
+
+				RemoveReference(&parent.Host, ref)
+			case *mo.ComputeResource:
+				ctx.Map.Remove(ctx, parent.Self)
+			}
+
+			c.Host = append(c.Host, ref)
+			host.Parent = &c.Self
```

Comment:
Is there an event of any kind that should trigger from the host movement?  I don't have a VC to hand - possibly:
* HostRemovedEvent
* HostNotInClusterEvent

Reply:
I didn't see those or other events triggered by MoveInto_Task using a real VC.